src/mysql-util/src/desc.rs

```diff
@@ -21,20 +29,139 @@ pub struct MySqlTableDesc {
+        }
+    }
+
+    fn from_proto(proto: ProtoMySqlTableDesc) -> Result<Self, TryFromProtoError> {
+        Ok(Self {
+            schema_name: proto.schema_name,
+            name: proto.name,
+            columns: proto
+                .columns
+                .into_iter()
+                .map(MySqlColumnDesc::from_proto)
+                .collect::<Result<_, _>>()?,
+        })
+    }
+}
+
+impl Arbitrary for MySqlTableDesc {
+    type Parameters = ();
+    type Strategy = BoxedStrategy<Self>;
+
+    fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
+        (
+            any::<String>(),
+            any::<String>(),
+            any::<Vec<MySqlColumnDesc>>(),
+        )
+            .prop_map(|(schema_name, name, columns)| Self {
+                schema_name,
+                name,
+                columns,
+            })
+            .boxed()
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
 pub struct MySqlColumnDesc {
     /// The name of the column.
     pub name: String,
     /// The MySQL datatype of the column.
-    pub data_type: MySqlDataType,
-    /// Whether the column is nullable.
-    pub nullable: bool,
+    pub column_type: ColumnType,
+    pub column_key: Option<MySqlColumnKey>,
     // TODO: add more column properties
 }
 
-#[derive(Debug, Clone, Eq, PartialEq)]
-pub enum MySqlDataType {
-    Int,
-    Varchar(usize),
-    // TODO: add more data types
+impl RustType<ProtoMySqlColumnDesc> for MySqlColumnDesc {
+    fn into_proto(&self) -> ProtoMySqlColumnDesc {
+        ProtoMySqlColumnDesc {
+            name: self.name.clone(),
+            column_type: Some(self.column_type.into_proto()),
+            column_key: self.column_key.into_proto(),
+        }
+    }
+
+    fn from_proto(proto: ProtoMySqlColumnDesc) -> Result<Self, TryFromProtoError> {
+        Ok(Self {
+            name: proto.name,
+            column_type: proto
+                .column_type
+                .into_rust_if_some("ProtoMySqlColumnDesc::column_type")?,
+            column_key: proto
+                .column_key
+                .map(MySqlColumnKey::from_proto)
+                .transpose()?,
+        })
+    }
+}
+
+impl Arbitrary for MySqlColumnDesc {
+    type Parameters = ();
+    type Strategy = BoxedStrategy<Self>;
+
+    fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
+        (
+            any::<String>(),
+            any::<ColumnType>(),
+            any::<Option<MySqlColumnKey>>(),
+        )
+            .prop_map(|(name, column_type, column_key)| Self {
+                name,
+                column_type,
+                column_key,
+            })
+            .boxed()
+    }
+}
+
+/// Refers to the COLUMN_KEY column of the information_schema.columns table
+/// https://dev.mysql.com/doc/refman/8.0/en/information-schema-columns-table.html
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub enum MySqlColumnKey {
+    PRI,
```

Comment:
After more investigation into mysql constraints I'm going to remove this `column_key` field as it's pretty useless since it doesn't tell provide enough information about multi-column indexes.

Instead I've figured out how to properly query the `information_schema.statistics` table to get the appropriate unique index info, and will represent those as `keys` on the `MySqlTableDesc` similar to how we do for Postgres